## Disciplina - Introdução a Ciência de Dados - Univesp
## Exercícios de Apoio da Semana 5


### 1 – Indique 3 algoritmos da biblioteca Scikit-Learn que podem ser utilizados tanto para regressão quanto para classificação.


1. teatea
2. asfasf
3. fasfsfaf

### 2 – Importe os dados de um arquivo CSV de avaliações de um modelo de Iphone, disponível no Kaggle (https://www.kaggle.com/kmldas/apple-iphone-se-reviews-ratings).  Uso de bibliotecas é por sua conta!



In [54]:
import pandas as pd
df_apple = pd.read_csv('APPLE_iPhone_SE.csv')
df_apple.head()

Ratings         Comment                                            Reviews
0        5          Super!  Great camera for pics and videos Battery life ...
1        5       Must buy!  Great device. Let me tell the Pros..1. Superb ...
2        5   Great product  Who all loves older size i.e., 4.7 inch type s...
3        5  Simply awesome  This iPhone SE is the best phone ever you get....
4        5  Classy product  This is my second iphone after iphone 4s. I’ve...

### 3 – Quantas linhas tem o arquivo?


In [55]:
df_apple.count()

Ratings    9713
Comment    9713
Reviews    9713
dtype: int64

**Resposta:** 9713 linhas

### 4 – Qual a média das notas atribuídas?


In [56]:
df_apple['Ratings'].mean()

4.4563986409966025

In [57]:
df_apple.describe(include= 'all')

Ratings    Comment        Reviews
count   9713.000000       9713           9713
unique          NaN        264           8229
top             NaN  Wonderful  GoodREAD MORE
freq            NaN        544            245
mean       4.456399        NaN            NaN
std        1.032911        NaN            NaN
min        1.000000        NaN            NaN
25%        4.000000        NaN            NaN
50%        5.000000        NaN            NaN
75%        5.000000        NaN            NaN
max        5.000000        NaN            NaN

**Resposta:** 4,456399

### 5 – Qual é a nota mais atribuída? Quantas vezes ela foi atribuída? Quantos % das notas totais?


In [60]:
groupby = df_apple.groupby('Ratings')['Ratings'].count()
df_groupby = pd.DataFrame({'Rating':groupby.index, 'QTDE':groupby.values})
QTDE_TT = df_groupby['QTDE'].sum()
df_groupby['Porcentagem'] = 100*df_groupby['QTDE']/QTDE_TT
df_groupby.sort_values(by='Porcentagem', ascending=False)

Rating  QTDE  Porcentagem
4       5  6788    69.885720
3       4  1717    17.677340
2       3   535     5.508082
0       1   474     4.880058
1       2   199     2.048801

**Resposta:** Rating 5 foi atribuído 6.788 vezes e representa 69,9% das notas totais

### 6 – Qual a nota menos atribuída? Qual a porcentagem que ela representa no todo?


**Resposta:** Rating 2 foi atribuído 199 vezes e representa 2,05% das notas totais

### 7 – Um dos comentários atribuídos é “Worth every penny” ou seja, “Vale cada centavo”. Faça a alteração nessas linhas do texto em inglês para o texto em português.

In [ ]:
df_apple[df_apple['Comment'].str.contains("Worth every penny")]

Ratings            Comment   
56          5  Worth every penny  \
115         5  Worth every penny   
123         5  Worth every penny   
140         5  Worth every penny   
161         5  Worth every penny   
...       ...                ...   
9609        5  Worth every penny   
9611        5  Worth every penny   
9643        5  Worth every penny   
9659        5  Worth every penny   
9672        5  Worth every penny   

                                                Reviews  
56    Best model if you wish to enter the Apple ecos...  
115   It's budget phone overall is the best only ban...  
123   Best phone i have ever purchased..tho battery ...  
140   Nice apple product. It works very fast and bes...  
161   Thumbs up to this new product from apple,just ...  
...                                                 ...  
9609  Overall good product but Some apps crashing on...  
9611  Except battery everything is perfect and wonde...  
9643  The phone looks awesome and is complete value ...  
9659  Phone with good performance and camera. Seamle...  
9672                        Good mobile in iOSREAD MORE  

[374 rows x 3 columns]

In [ ]:
df_apple['Comment'].replace('Worth every penny', 'Vale cada centavo', inplace=True)
df_apple[df_apple['Comment'].str.contains("Vale cada centavo")]

Ratings            Comment   
56          5  Vale cada centavo  \
115         5  Vale cada centavo   
123         5  Vale cada centavo   
140         5  Vale cada centavo   
161         5  Vale cada centavo   
...       ...                ...   
9609        5  Vale cada centavo   
9611        5  Vale cada centavo   
9643        5  Vale cada centavo   
9659        5  Vale cada centavo   
9672        5  Vale cada centavo   

                                                Reviews  
56    Best model if you wish to enter the Apple ecos...  
115   It's budget phone overall is the best only ban...  
123   Best phone i have ever purchased..tho battery ...  
140   Nice apple product. It works very fast and bes...  
161   Thumbs up to this new product from apple,just ...  
...                                                 ...  
9609  Overall good product but Some apps crashing on...  
9611  Except battery everything is perfect and wonde...  
9643  The phone looks awesome and is complete value ...  
9659  Phone with good performance and camera. Seamle...  
9672                        Good mobile in iOSREAD MORE  

[372 rows x 3 columns]

In [ ]:
df_apple[df_apple['Comment'].str.contains("Worth every penny")]

Ratings                        Comment   
3471        5  Wonderful - Worth every penny  \
7212        5             Worth every penny!   

                                                Reviews  
3471  Best budget iPhone Apple released. Prior to th...  
7212  Bought 128GB version for 30k, worth every penn...

### 8 – Exclua todas as linhas em que o Comentário é “Hated it!”. Esse comentário aparece 23 vezes na lista de comentários.

In [ ]:
df_apple[df_apple['Comment'].str.contains("Hated it!")]

Ratings                                            Comment   
548         1  Hated it! Charger and ear phone not included i...  \
905         1                                          Hated it!   
934         1                                          Hated it!   
941         1                                          Hated it!   
1656        1                                          Hated it!   
1690        1                                          Hated it!   
1738        1                                          Hated it!   
2006        1                                          Hated it!   
3091        1                                          Hated it!   
3246        1                                          Hated it!   
3425        1                                          Hated it!   
3641        1                                          Hated it!   
3646        1                                          Hated it!   
3897        1                                          Hated it!   
3930        1                                          Hated it!   
3939        1                                          Hated it!   
4477        1                                          Hated it!   
5376        1                                          Hated it!   
5386        1                                          Hated it!   
6852        1                                          Hated it!   
7257        1                                          Hated it!   
8516        1                                          Hated it!   
8755        1                                          Hated it!   
9288        1                                          Hated it!   

                                                Reviews  
548   Charger and ear phone not included in this mod...  
905   One of the worst mobile phones I have ever use...  
934   If any one of you want to exchange your phone ...  
941   While buying it was written that earpod and ch...  
1656  i you have extra money then Apple SE best for ...  
1690                           Battary problemREAD MORE  
1738  Camera is great but phone is not worth. Batter...  
2006                              Varry bad ☹️READ MORE  
3091  Waste of money no charging adaptor no earphone...  
3246  Flipkart give worst product it is also not wat...  
3425     Got it without charger and headphonesREAD MORE  
3641                          Worst ever deal.READ MORE  
3646  Wrong charging cable. box was manipulated item...  
3897  I recently purchased this phone with complete ...  
3930            Have a damages in display edgeREAD MORE  
3939  the product is not working after using 5 month...  
4477  It's the worst phone don't buy it...my battery...  
5376                               Not so goodREAD MORE  
5386                         Very poor qualityREAD MORE  
6852                          Vary bad batteryREAD MORE  
7257                              Poor batteryREAD MORE  
8516  I would never recommend to purchase anything f...  
8755  Worst battery for this phone.Seems somthing fa...  
9288          Sound is not coming loud on callREAD MORE

In [ ]:
df_apple.drop(df_apple[df_apple['Comment'].str.contains("Hated it!")].index, inplace=True)
print(df_apple.count())
df_apple[df_apple['Comment'].str.contains("Hated it!")]

Ratings    9689
Comment    9689
Reviews    9689
dtype: int64


Empty DataFrame
Columns: [Ratings, Comment, Reviews]
Index: []